content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/redhat_customizerequest.rb

```diff
@@ -3,226 +3,198 @@
+                  @mapping = false
+                end
+
+                def main
+                  log(:info, "Provision:<#{prov.id}> Request:<#{prov.miq_provision_request.id}> Type:<#{prov.type}>")
+                  log(:info, "Template: #{template.name} Provider: #{provider.name} Vendor: #{template.vendor} Product: #{product}")
+                  # Build case statement to determine which type of processing is required
+                  case prov.type
+                  when 'ManageIQ::Providers::Redhat::InfraManager::Provision'
+                    process_redhat if @mapping
+                  when 'ManageIQ::Providers::Redhat::InfraManager::ProvisionViaIso'
+                    process_redhat_iso if @mapping
+                  when 'ManageIQ::Providers::Redhat::InfraManager::ProvisionViaPxe'
+                    process_redhat_pxe if @mapping
+                  else
+                    log(:info, "Provisioning Type: #{prov.type} does not match, skipping processing")
+                  end
+                end
+
+                private
+
+                def prov
+                  @prov ||= @handle.root["miq_provision"].tap do |req|
+                    raise 'miq_provision not specified' if req.nil?
+                  end
+                end
+
+                def template
+                  @template ||= prov.vm_template.tap do |req|
+                    raise 'vm_template not specified' if req.nil?
+                  end
+                end
+
+                def provider
+                  @provider ||= template.ext_management_system.tap do |ems|
+                    raise 'ext_management_system not specified' if ems.nil?
+                  end
+                end
+
+                def product
+                  @product ||= template.operating_system&.product_name&.downcase
+                end
+
+                def ws_values
+                  @ws_values ||= prov.options.fetch(:ws_values, {})
+                end
+
+                def log(level, msg, update_message = false)
+                  @handle.log(level, msg)
+                  @handle.root['miq_provision'].message = msg if @handle.root['miq_provision'] && update_message
+                end
+
+                def update_and_raise(msg)
+                  @handle.root['miq_provision'].message = msg
+                  raise msg
+                end
+
+                def process_redhat_pxe
+                  if product.include?("windows")
+                    # find the windows image that matches the template name if a PXE Image was NOT chosen in the dialog
+                    if prov.get_option(:pxe_image_id).nil?
+
+                      log(:info, "Inspecting prov.eligible_windows_images: #{prov.eligible_windows_images.inspect}")
+                      pxe_image = prov.eligible_windows_images.detect { |pi| pi.name.casecmp(template.name) == 0 }
+                      if pxe_image.nil?
+                        update_and_raise("Failed to find matching PXE Image")
```

Comment:
The behavior before is to log in the file, update the object then raise an error. 
Seems logging in the file is skipped here. Is it by intention?